src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -205,13 +210,26 @@ public DatasetTerm toDatasetTerm(Dataset dataset) {
     DatasetTerm term = new DatasetTerm();
 
     term.setDatasetId(dataset.getDataSetId());
+    if (Objects.nonNull(dataset.getCreateUserId())) {
```

Comment:
This is the convention of this code, but I'm wondering why the code doesn't compare against `null` directly. To me using `==` is easier to read than a function call but that is a subjective difference.
```suggestion
    if (dataset.getCreateUserId() != null) {
```

Alternatively, you can avoid repeating the expression `dataset.getCreateUserId()` by using `Optional`. That would look like:
```
    Optional.ofNullable(dataset.getCreateUserId()).ifPresent(
        userId -> {
          User user = userDAO.findUserById(userId);
          term.setCreateUserId(dataset.getCreateUserId());
          term.setCreateUserDisplayName(user.getDisplayName());
        }
    );
```

Reply:
I do like the optional approach in this case, thanks for the tip!